packages/@aws-cdk/aws-codepipeline-actions/test/integ.pipeline-ecs-import-cross-account.lit.ts

```diff
@@ -0,0 +1,208 @@
+            'iam:PassedToService': [
+              'ec2.amazonaws.com',
+              'ecs-tasks.amazonaws.com',
+            ],
+          },
+        },
+      }),
+    );
+  }
+}
+
+/**
+ * This is the Stage which does our import using {@link TestImportEcsStack}.
+ */
+class TestImportEcsStage extends cdk.Stage {
+  public readonly serviceArn: string;
+  public readonly clusterName: string;
+  public readonly deployRole: iam.IRole;
+
+  constructor(scope: Construct, id: string, props: cdk.StageProps) {
+    super(scope, id, props);
+    const testStack = new TestImportEcsStack(this, 'ecsStack', {});
+    this.serviceArn = testStack.serviceArn;
+    this.clusterName = testStack.clusterName;
+    this.deployRole = testStack.deployRole;
+  }
+}
+
+/**
+ * This is our pipeline which will import our existing ECS Service and deploy
+ * to is using {@link EcsDeployAction} using our {@link TestImportEcsStage}
+ */
+class PipelineEcsDeployImportStack extends cdk.Stack {
+  constructor(scope: Construct, id: string, props: cdk.StackProps) {
+    super(scope, id, props);
+    const artifact = new codepipeline.Artifact('Artifact');
+    const bucket = new s3.Bucket(this, 'PipelineBucket', {
+      versioned: true,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+    });
+    const source = new cpactions.S3SourceAction({
+      actionName: 'Source',
+      output: artifact,
+      bucket,
+      bucketKey: 'key',
+    });
+    const pipeline = new codepipeline.Pipeline(this, 'Pipeline', {
+      stages: [
+        {
+          stageName: 'Source',
+          actions: [source],
+        },
+      ],
+    });
+    const testStage = new TestImportEcsStage(
+      this,
+      'TestStage',
+      {
+        env: {
+          region: 'us-west-2',
+          account: '123456789012',
+        },
+      },
+    );
+    const testIStage = pipeline.addStage(testStage);
+    /**
+     * This imports the service so it can be used by the ECS Deploy Action.
+     * This must use the serviceArn so that the region is set correctly.
+     * The VPC doesn't actually matter and it doesn't get created.
+     * The construct ids will need to be unique.
+    */
+    const service = ecs.FargateService.fromFargateServiceAttributes(this, 'FargateService', {
+      serviceArn: testStage.serviceArn,
+      cluster: ecs.Cluster.fromClusterAttributes(this, 'Cluster', {
+        vpc: new ec2.Vpc(this, 'Vpc'),
+        securityGroups: [],
+        clusterName: testStage.clusterName,
+      }),
+    });
+    /**
+     * It is highly recommended passing in the role from the stage/stack, if not a new role
+     * will be added to the pipeline action, will not exist in the account you are deploying to.
+     */
+    const deployAction = new cpactions.EcsDeployAction({
+      actionName: 'ECS',
+      service: service,
+      imageFile: artifact.atPath('imageFile.json'),
+      role: testStage.deployRole,
+    });
+    testIStage.addAction(deployAction);
+  }
+}
```

Comment:
I'm curious - is there *any* difference in the pipeline Stacks between the two examples? I understand that the service Stack is different, but I *think* the Pipeline Stacks are the same...? If that's true, can we remove the duplication between the examples? Either merge them to one example that uses the same PipelineStack, but two different service Stacks, or keep two examples, but extract the PipelineStack into a separate class outside of them.

Reply:
This is probably a matter of how verbose we want the examples to be, the difference is just the stages and stacks.

You thinking have a pipeline which creates and deploys in one stage and another stage which imports a different different service to deploy to?

We could probably do that with comments.

I think we can hide parts of the code from the readme as well right?

I could have the pipeline create in a different class and import it and set the pipeline based on that to show the add of the stack with the action.   Which would reduce the somewhat boilerplate pipeline aspects.

Reply:
> You thinking have a pipeline which creates and deploys in one stage and another stage which imports a different different service to deploy to?

No. I meant having one `PipelineStack` that takes something (maybe a `cdk.Stage`, maybe something else, like a new interface containing `serviceArn`, `clustrerName`, etc. properties?) as the argument, and then doing the correct thing with the passed argument.

Reply:
I think I have addressed some of this duplication.
The big item we want to make sure we have in the example is the fact the you have to import the service again in the pipeline stack.  Which if we pass the stage into the pipelinestack, it would be some what hidden to the documentation/example.